src/js/utils/hooks.ts

```diff
@@ -91,3 +92,52 @@ export function useUrlSearchParams(key: string, defaultValue?: string): [string,
 
     return [value, (value: string) => updateUrlSearchParams(value, key, history)];
 }
+
+/**
+ * Updates the URL search parameters by either adding a new value for a given key or removing the key-value pair
+ *
+ * @param value - The value to be used in the search parameter
+ * @param key - The search parameter key to be managed
+ * @param history - The history object
+ */
+function updateUrlSearchParamsList(value: string, key: string, history: HistoryType) {
+    const params = new URLSearchParams(window.location.search);
+    let values = params.getAll(key);
+
+    if (value && !values.includes(value.toString())) {
+        params.append(key, value);
+    } else {
+        values = xor(values, [value.toString()]);
+        params.delete(key);
+        values.forEach(val => params.append(key, val));
+    }
+
+    history?.replace({
+        pathname: window.location.pathname,
+        search: params.toString() ? `?${params.toString()}` : null,
+    });
+}
+
+/**
+ * Hook for managing and synchronizing a list of URL search parameters with a component's state
+ *
+ * @param key - The search parameter key to be managed
+ * @param defaultValue - The default values to use when the search parameter key is not present in the URL
+ * @returns Object - An object containing the current values and a function to set the URL search parameter
+ */
+export function useUrlSearchParamsList(key: string, defaultValue?: string[]): [string[], (newValue: string) => void] {
+    const history = useHistory();
+    const location = useLocation();
+    const firstRender = useRef(true);
+
+    let value = new URLSearchParams(location.search).getAll(key);
+
+    if (firstRender.current && defaultValue && !value.length) {
+        value = defaultValue;
+        value.forEach(val => updateUrlSearchParamsList(val, key, history));
```

Comment:
Why not just pass the whole array? Seems like it would simplify the logic quite a bit and be faster